mediatype/mediatype.go

```diff
@@ -482,11 +482,32 @@ func fixNewlines(value string) string {
 	return value
 }
 
-// fixUnexpectedTag removes an unexpected <!DOCTYPE tag
-func fixUnexpectedTag(value string) string {
-	dtIdx := strings.Index(value, "<")
-	if dtIdx > -1 {
-		return value[0:dtIdx]
+// removeTrailingHTMLTags removes an unexpected HTML tags at the end of media type.
+func removeTrailingHTMLTags(value string) string {
+	tagStart := 0
+	closeTags := 0
+
+loop:
+	for i := len(value) - 1; i > 0; i-- {
+		c := value[i]
+		switch c {
+		case '"':
+			if closeTags == 0 { // quotes started outside the tag, aborting
```

Comment:
Is this needed - I mean do we want to stop on the last quote outside of a tag?

Reply:
we don't want to remove tags inside quotes, e.g. `multipart/alternative; boundary="<myboundary>"` having valid boundary